inject-groovy/src/main/groovy/io/micronaut/ast/groovy/visitor/GroovyClassElement.java

```diff
@@ -761,6 +766,19 @@ private List<AnnotatedNode> getEnclosedElements(ClassNode classNode, ElementQuer
             }
         }
 
+        private boolean isNonAbstract(ClassNode classNode, MethodNode methodNode) {
+            if (methodNode.isDefault()) {
+                return false;
+            }
+            if (methodNode.isPrivate() && classNode.isInterface()) {
+                return false;
+            }
+            if (!methodNode.isAbstract() && classNode.isInterface()) {
```

Comment:
i dont understand this method. private & interface -> abstract? !abstract & interface -> abstract?

Reply:
private methods are never abstract, and the other one is the default method

Reply:
this code says that private interface methods *are* abstract though

Reply:
Ah I see, fixed

Reply:
can drop the interface check too i guess